frontend/src/components/Signup.js

```diff
@@ -5,50 +5,155 @@ class Signup extends Component {
 
     constructor(props) {
         super(props);
-        this.state = { isVolunteer: false };
+        this.state = { 
+            userType: "",
+            isAuthenticated: false,
+            personalData: { //shared across all users
+                firstName: "",
+                lastName: "",
+                email: "",
+                password: "",
+                site: ""
+            },
+            volunteerData: { //volunteers only
+                driver: false,
+                kitchenStaff: false,
+                isAuthenticated_driver: false,
+                isAuthenticated_kitchenStaff: false,
+            }
+         };
     }
     
-    isNotVolunteer = () => {
-        this.setState({isVolunteer: false});
+    // updates personal data
+    handleChange = (event) => {
+        let personalData = this.state.personalData;
+        personalData[event.target.id] = event.target.value;
+
+        this.setState({personalData: personalData});
+    }
+
+    changeUserType = (event) => {
+        this.setState({userType: event.target.value});
+    }
+
+    changeVolunteerType = (event) => {
+        let volunteerData = this.state.volunteerData;
+        if (event.target.value === "driver") {
```

Comment:
A user should have the option to be both a driver and a kitchen staff. It's not exclusive